libcxx/test/std/ranges/range.adaptors/range.stride.view/ctad.pass.cpp

```diff
@@ -0,0 +1,96 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// template <class _Range>
+// stride_view(_Range&&, range_difference_t<_Range>) -> stride_view<views::all_t<_Range>>;
+
+#include "types.h"
+#include <concepts>
+#include <ranges>
+
+struct View : std::ranges::view_base {
+  int* begin() const;
+  int* end() const;
+};
+
+struct Range {
+  int* begin() const;
+  int* end() const;
+};
+
+constexpr bool testCTAD() {
```

Comment:
These CTAD tests should be testing CTAD only, not the iterator operations. So something like this should be sufficient:

```c++
int a[] = {1, 2, 3, 4, 5};
using Range = BasicTestRange<cpp17_input_iterator<int*>>;
using View = BasicTestView<int*>;

{
  Range  range(a, a + 5);
  std::ranges::stride_view view(range, 2);
  ASSERT_SAME_TYPE(decltype(view), std::ranges::stride_view<std::ranges::ref_view<Range>>);
}
{
  Range  range(a, a + 5);
  std::ranges::stride_view view(std::move(range), 2);
  ASSERT_SAME_TYPE(decltype(view), std::ranges::stride_view<std::ranges::owning_view<Range>>);
}
{
  Range range(a, a + 5);
  View  v(range.begin(), range.end());
  std::ranges::stride_view view(v, 2);
  ASSERT_SAME_TYPE(decltype(view), std::ranges::stride_view<View>);
}
```

I think the above tests are both necessary and sufficient for what we're trying to test here.